### PR TITLE
Add bounds to all range related operations

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
This PR ensure that range related operations (propagation of strings, evidence construction in sinks) take an upper bound regarding the number of elements they can iterate and the number of spans they can generate. 

# Motivation
When dealing with huge string manipulation operations (e.g. joining a high number of strings), IAST might become a bottleneck due to the need to analyze each string separately.

### CODE REVIEW

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java [hunk 1]
```diff
@@ -153,38 +149,26 @@ public void onStringConcatFactory(
     }
 
     final TaintedObjects taintedObjects = ctx.getTaintedObjects();
-    final Map<Integer, Range[]> sourceRanges = new HashMap<>();
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java:156]: Range builder is a specialized data structure designed to work with ranges with low allocation overhead, it should provide better time/space complexity compared to do a previous lookup of tainted ranges + storage in a map.

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/LdapInjectionModuleImpl.java [hunk 2]
```diff
@@ -26,16 +23,8 @@ public void onDirContextSearch(
...
-      LOG.debug("No IastRequestContext available");
-      return;
-    }
-    final TaintedObjects to = ctx.getTaintedObjects();
     checkInjection(
         VulnerabilityType.LDAP_INJECTION,
-        rangesProviderFor(to, name),
-        rangesProviderFor(to, filterExpr),
-        rangesProviderFor(to, filterArgs));
+        Iterators.join(Iterators.of(name, filterExpr), Iterators.of(filterArgs)));
```
Comment (jandro996) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/LdapInjectionModuleImpl.java:28]: This should be like this?

`Iterators.join(Iterators.of(name), Iterators.of (filterExpr), Iterators.of(filterArgs)));`

**Review by jandro996 — APPROVED** [review]
Great Job! these changes will improve our delivery pace for new Vulnerabilities 👍

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/LdapInjectionModuleImpl.java (hunk 2 above)
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-iast/src/main/java/com/datadog/iast/sink/LdapInjectionModuleImpl.java:28]: You save on `Iterator` allocation by joining `name` and `filterExpr` in a single one.

**Pull request merged**